src/btf_parser.cpp

```diff
@@ -1081,3 +1085,393 @@ std::map<std::string, size_t> parse_btf_map_sections(const btf_type_data& btf_da
     }
     return map_offset_to_descriptor_index;
 }
+
+/**
+ * @brief Write a type into a vector and grow it.
+ *
+ * @tparam T Type to write
+ * @param[in] btf The .BTF section
+ * @param[in] value The value to write
+ */
+template <typename T>
+void _write_btf(std::vector<std::byte>& btf, const T& value) {
```

Comment:
Clang-Tidy warns about using identifier that is reserved in the global namespace.
Better: `static void write_btf(...);`

Reply:
(Similarly for all other function names that start with an underscore, here and elsewhere).